### PR TITLE
Connect beta

### PR DESCRIPTION
**Pull request by SteveMieskoski**

### CODE REVIEW

**Review by gamalielhere — APPROVED** [review]

File: app/scripts/abiDefinitions/ropstenAbi.json [hunk 1]
```diff
@@ -36,14 +36,14 @@
...
 },{
     "name": "Mock ELF Token",
-    "address": "0x7174FCb9C2A49c027C9746983D8262597b5EcCb1",
+    "address": "0x9Fcc27c7320703c43368cf1A4bf076402cd0D6B4",
```
Comment (kvhnuke) [inline app/scripts/abiDefinitions/ropstenAbi.json]: are these real token contracts?

File: app/scripts/controllers/signMsgCtrl.js [hunk 2]
```diff
@@ -144,7 +144,43 @@ var signMsgCtrl = function($scope, $sce, walletService) {
...
+              //   $scope.notifier.success('Successfully Signed Message with ' + $scope.wallet.getAddressString());
+              //   // mewConnect.disconnectRTCDirect();
+              // });
+              //TODO hash message before send.  Currently sending as plain text
+              connectApp.signMessage(thisMessage);
+
+              $scope.$on("$destroy", function () {
+                // $scope.mewConnect.disconnectRTCDirect();
+                // // if(MewConnect.instance){
+                // //     console.log("decryptWalletCtrl:363", MewConnect); //todo remove dev item
```
Comment (kvhnuke) [inline app/scripts/controllers/signMsgCtrl.js]: ?

File: app/scripts/controllers/signMsgCtrl.js [hunk 3]
```diff
@@ -144,7 +144,43 @@ var signMsgCtrl = function($scope, $sce, walletService) {
...
+                    // mewConnect.disconnectRTCDirect();
+                })
+
+
+              // mewConnect.setMessageSignerCallback(function (data, next) {
+              //   $scope.signMsg.signedMsg = JSON.parse(data);
+              //   $scope.notifier.success('Successfully Signed Message with ' + $scope.wallet.getAddressString());
+              //   // mewConnect.disconnectRTCDirect();
+              // });
+              //TODO hash message before send.  Currently sending as plain text
```
Comment (kvhnuke) [inline app/scripts/controllers/signMsgCtrl.js]: good point!

File: app/scripts/controllers/txStatusCtrl.js [hunk 4]
```diff
@@ -34,8 +34,8 @@ var txStatusCtrl = function($scope) {
     var txToObject = function(tx) {
         var txStatus = $scope.txStatus;
         if (tx) {
-            console.log('txToObject')
-            console.log(tx)
+            // console.log('txToObject')
```
Comment (kvhnuke) [inline app/scripts/controllers/txStatusCtrl.js]: take these out please!

File: app/scripts/main.js [hunk 5]
```diff
@@ -1,176 +1,187 @@
...
+var bity = require('./bity')
+window.bity = bity
+var kyber = require('./kyber')
+window.kyber = kyber
+var ens = require('./ens')
+window.ens = ens
+var domainsale = require('./domainsale')
+window.domainsale = domainsale
+var translate = require('./translations/translate.js')
+var socketIo = require('./staticJS/socketIO.min') // using the npm version breaks mewConnect
```
Comment (kvhnuke) [inline app/scripts/main.js:62]: any specific reason why?

File: app/scripts/main.js [hunk 6]
```diff
@@ -1,176 +1,187 @@
...
+  var trezorConnect = require('./staticJS/trezorConnect')
+  var digitalBitboxUsb = require('./staticJS/digitalBitboxUsb')
+  var digitalBitboxEth = require('./staticJS/digitalBitboxEth')
+  var secalotUsb = require('./staticJS/secalotUsb')
+  var secalotEth = require('./staticJS/secalotEth')
+
+  require('./staticJS/adapter') // adapter to ensure a common api for webRTC
+  var MewConnectEth = require('./staticJS/mewConnectEth')
+  var MewConnect = require('./staticJS/MewConnect.min').Initiator
+  var MewConnectCrypto = require('./staticJS/MewConnect.min').Crypto
```
Comment (kvhnuke) [inline app/scripts/main.js]: i dont think we are using this?

File: app/scripts/main.js [hunk 7]
```diff
@@ -1,176 +1,187 @@
...
+  var ledgerEth = require('./staticJS/ledger-eth')
+  var trezorConnect = require('./staticJS/trezorConnect')
+  var digitalBitboxUsb = require('./staticJS/digitalBitboxUsb')
+  var digitalBitboxEth = require('./staticJS/digitalBitboxEth')
+  var secalotUsb = require('./staticJS/secalotUsb')
+  var secalotEth = require('./staticJS/secalotEth')
+
+  require('./staticJS/adapter') // adapter to ensure a common api for webRTC
+  var MewConnectEth = require('./staticJS/mewConnectEth')
+  var MewConnect = require('./staticJS/MewConnect.min').Initiator
```
Comment (kvhnuke) [inline app/scripts/main.js:90]: which library is this? is there a npm version available?

Comment (SteveMieskoski) [reply app/scripts/main.js:90]: It is the MEWconnect client. I will publish it as an npm package.

**Comment by ghost** [comment]
Great feature, I would like to test them when it is live 👍

File: app/scripts/main.js (hunk 5 above)
Comment (SteveMieskoski) [reply app/scripts/main.js:62]: wasn't able to get to the bottom of it.  The version from npm would send the messages with a different type identifier versus the stand alone (type 5 vs type 2.  Or at least that is the most direct difference I noticed.  It is a point for further investigation.

**Pull request merged**